nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/composite/CompositeUtil.java

```diff
@@ -637,50 +637,41 @@ public Locale getLocale(MultivaluedMap<String,String> requestHeaders) {
         try {
             Enumeration<URL> urls = similarClass.getClassLoader().getResources("META-INF/restmodelextensions");
             while (urls.hasMoreElements()) {
                 URL url = urls.nextElement();
-                reader = new BufferedReader(new InputStreamReader(url.openStream()));
-                while (reader.ready()) {
-                    final String line = reader.readLine();
-                    if ((line == null) || line.isEmpty()) {
-                        continue;
-                    }
-                    if (line.charAt(0) != '#') {
-                        if (!line.contains(":")) {
-                            RestLogging.restLogger.log(Level.INFO,
-                                    RestLogging.INCORRECTLY_FORMATTED_ENTRY,
-                                    new String[]{
-                                        "META-INF/restmodelextensions",
-                                        line
-                                    });
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
+                    while (reader.ready()) {
+                        final String line = reader.readLine();
+                        if ((line == null) || line.isEmpty()) {
+                            continue;
                         }
-                        String[] entry = line.split(":");
-                        String base = entry[0];
-                        String ext = entry[1];
-                        List<String> list = modelExtensions.get(base);
-                        if (list == null) {
-                            list = new ArrayList<String>();
-                            modelExtensions.put(base, list);
+                        if (line.charAt(0) != '#') {
+                            if (!line.contains(":")) {
+                                RestLogging.restLogger.log(Level.INFO,
+                                        RestLogging.INCORRECTLY_FORMATTED_ENTRY,
+                                        new String[]{
+                                                "META-INF/restmodelextensions",
+                                                line
+                                });
+                            }
+                            String[] entry = line.split(":");
+                            String base = entry[0];
+                            String ext = entry[1];
+                            List<String> list = modelExtensions.get(base);
+                            if (list == null) {
+                                list = new ArrayList<>();
+                                modelExtensions.put(base, list);
+                            }
+                            list.add(ext);
                         }
-                        list.add(ext);
                     }
                 }
-
             }
         } catch (IOException ex) {
             RestLogging.restLogger.log(Level.SEVERE, null, ex);
-        } finally {
```

Comment:
I think this block wasn't just "old style" it was also buggy because looking closely the resource is created in the outer `while` loop but this will only close the last resource from the loop.